homeassistant/components/smartthings/sensor.py

```diff
@@ -1127,80 +1142,73 @@ async def async_setup_entry(
-                                        component == MAIN
-                                        and description.exists_fn(
-                                            device.status[MAIN][capability][attribute]
-                                        )
-                                    )
-                                )
-                                and (
-                                    component == MAIN
-                                    or (
-                                        description.component_fn is not None
-                                        and description.component_fn(component)
-                                    )
-                                )
-                            ):
-                                if (
-                                    description.deprecated
-                                    and (
-                                        deprecation_info := description.deprecated(
-                                            device.status[MAIN]
-                                        )
-                                    )
-                                    is not None
-                                ):
-                                    version, reason = deprecation_info
-                                    if deprecate_entity(
-                                        hass,
-                                        entity_registry,
-                                        SENSOR_DOMAIN,
-                                        f"{device.device.device_id}_{MAIN}_{capability}_{attribute}_{description.key}",
-                                        f"deprecated_{reason}",
-                                        version,
-                                    ):
-                                        entities.append(
-                                            SmartThingsSensor(
-                                                entry_data.client,
-                                                device,
-                                                description,
-                                                MAIN,
-                                                capability,
-                                                attribute,
-                                            )
-                                        )
-                                    continue
-                                entities.append(
-                                    SmartThingsSensor(
-                                        entry_data.client,
-                                        device,
-                                        description,
-                                        component,
-                                        capability,
-                                        attribute,
-                                    )
-                                )
+        # Handle deprecated sensors
+        if (
+            description.deprecated
```

Comment:
Should we add `component == MAIN` condition here? Otherwise, the same entities may be created twice.